pkg/distributor/ha_tracker.go

```diff
@@ -453,20 +455,27 @@ func (h *defaultHaTracker) updateKVStoreAll(ctx context.Context, now time.Time)
 			}
 			var replica string
 			var receivedAt int64
+			var sampleTime time.Time
 			if uh.withinUpdateTimeout(now, entry.electedLastSeenTimestamp) {
 				// We have seen the elected replica recently; carry on with that choice.
 				replica = entry.elected.Replica
 				receivedAt = entry.electedLastSeenTimestamp
+				sampleTime = now
 			} else if uh.withinUpdateTimeout(now, entry.nonElectedLastSeenTimestamp) {
 				// Not seen elected but have seen another: attempt to fail over.
 				replica = entry.nonElectedLastSeenReplica
 				receivedAt = entry.nonElectedLastSeenTimestamp
+				if uh.useSampleTimeForFailover {
```

Comment:
why is the `if` here as well? i mean `updateKVStore` already checks the same `if` and uses `sampleTime` if necessary.